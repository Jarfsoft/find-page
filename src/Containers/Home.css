.App {
  text-align: center;
  padding-bottom: 2rem;
  display: grid;
  grid-template-columns: repeat(4, auto);
}

@media (max-width: 650px) {
  .App {
    grid-template-columns: auto;
  }
}
