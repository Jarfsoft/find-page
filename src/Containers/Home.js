import { React, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import actions from '../Actions';

import getDevices from '../deviceAPI';
import Device from '../Components/Device';
import './Home.css';

function Home() {
  const dispatch = useDispatch();
  const devices = useSelector((state) => state.devices);
  dispatch(actions.page('Devices'));
  const setDevices = (data) => {
    dispatch(actions.devices(data));
  };
  useEffect(() => {
    getDevices(setDevices);
  }, []);
  return (
    <div className="App">
      {devices ? devices.map((d) => (
        <Device
          key={d.name}
          imageUrl={d.image_url}
          id={d.id}
          name={d.name}
          price={d.price}
        />
      )) : <div className="loading" />}
    </div>
  );
}

export default Home;
