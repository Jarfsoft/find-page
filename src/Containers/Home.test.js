import { render, screen } from '@testing-library/react';
import Home from './Home';

test('', () => {
  render(<Home />);
});
