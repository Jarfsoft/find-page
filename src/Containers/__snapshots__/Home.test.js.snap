// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home render Snapshot check should render Home section 1`] = `
<div
  className="App"
/>
`;
